### PR TITLE
Allow configuring Windows service name, description and user

### PR DESCRIPTION
**Pull request by detouched**
This adds support for more environment variables to configure Windows Elasticsearch service:
- `SERVICE_DISPLAY_NAME` and `SERVICE_DESCRIPTION` to customize how the service is listed in the services list
- `SERVICE_USERNAME` and `SERVICE_PASSWORD` to configure the user to run the service (which will only be applied when both are set)

They are all optional and the current defaults will be used if they are not set.

### CODE REVIEW

File: distribution/src/main/resources/bin/service.bat [hunk 1]
```diff
@@ -162,7 +162,16 @@ if not "%ES_JAVA_OPTS%" == "" set JVM_OPTS=%JVM_OPTS%;%JVM_ES_JAVA_OPTS%
 if "%ES_START_TYPE%" == "" set ES_START_TYPE=manual
 if "%ES_STOP_TIMEOUT%" == "" set ES_STOP_TIMEOUT=0
 
-"%EXECUTABLE%" //IS//%SERVICE_ID% --Startup %ES_START_TYPE% --StopTimeout %ES_STOP_TIMEOUT% --StartClass org.elasticsearch.bootstrap.Elasticsearch --StopClass org.elasticsearch.bootstrap.Elasticsearch --StartMethod main --StopMethod close --Classpath "%ES_CLASSPATH%" --JvmSs %JVM_SS% --JvmMs %JVM_XMS% --JvmMx %JVM_XMX% --JvmOptions %JVM_OPTS% ++JvmOptions %ES_PARAMS% %LOG_OPTS% --PidFile "%SERVICE_ID%.pid" --DisplayName "Elasticsearch %ES_VERSION% (%SERVICE_ID%)" --Description "Elasticsearch %ES_VERSION% Windows Service - http://elasticsearch.org" --Jvm "%%JAVA_HOME%%%JVM_DLL%" --StartMode jvm --StopMode jvm --StartPath "%ES_HOME%" ++StartParams start
+if "%SERVICE_DISPLAY_NAME%" == "" set SERVICE_DISPLAY_NAME="Elasticsearch %ES_VERSION% (%SERVICE_ID%)"
+if "%SERVICE_DESCRIPTION%" == "" set SERVICE_DESCRIPTION="Elasticsearch %ES_VERSION% Windows Service - http://elasticsearch.org"
```
Comment (jasontedor) [inline distribution/src/main/resources/bin/service.bat]: Can you use `https` instead of `http`, and make the URL `elastic.co`?

Comment (detouched) [inline distribution/src/main/resources/bin/service.bat]: Sure, updated the URL

Comment (jasontedor) [inline distribution/src/main/resources/bin/service.bat]: Thanks. I'll ping someone from the Windows team to take a look at the rest.

**Comment by jasontedor** [comment]
This still needs a CLA to be signed but can you take a look @Mpdreamz?

**Comment by detouched** [comment]
@jasontedor I should be listed in a corporate CLA signed by Atlassian.

**Comment by Mpdreamz** [comment]
LGTM but I can't take it for a quick spin until Monday, @gmarz can you test this one today?

**Comment by gmarz** [comment]
Hey @detouched I just tried testing this and I'm hitting a few issues.

**1)** The service name doesn't properly handle spaces.

```
set SERVICE_DISPLAY_NAME=My Elasticsearch Service
```

results in the service name being installed as `My`

If I set it using quotes

```
set SERVICE_DISPLAY_NAME="My Elasticsearch Service"
```

The following error occurs during installation: `Elasticsearch was unexpected at this time.`

**2)** The service description when set doesn't appear to be taking.  It's installed with an empty description.

**3)** The username and password also do not seem to be taking.  It still uses the local system account.

The default values work, however.

**Comment by detouched** [comment]
Hello @gmarz, thank you for your feedback!

**1.** Indeed, with the syntax you used spaces are handled incorrectly. I used `set "KEY=Value with spaces"` instead but I agree that it's not a common way to set environment variables. So I've just updated the PR with a fix for that. Now syntax _without_ quotes should work (I believe it's how values with spaces are usually set):
```
set SERVICE_DISPLAY_NAME=My Elasticsearch Service
set SERVICE_DESCRIPTION=Elasticsearch Service Description
```
**2.** Unfortunately I can't reproduce that — it works for me, similar to the display name. Maybe it was also due to the fix?
**3.** The username should be fully qualified (i.e. with the domain), and such user should exist in the system. So, for instance, in my case, any of following will work (assuming `SERVICE_PASSWORD` is set):
```
set SERVICE_USERNAME=.\dpenkin
```
or
```
set SERVICE_USERNAME=DESKTOP-ETHEERQ\dpenkin
```

**Comment by gmarz** [comment]
LGTM.  I've tested your fixes and all looks good. :+1: 

> The username should be fully qualified (i.e. with the domain), and such user should exist in the system. 

Yup, that was my issue.

The empty description seems to be fixed now as well.  It was probably related to the missing quotes.  

Thanks for the quick turnaround!

**Comment by jasontedor** [comment]
> I should be listed in a corporate CLA signed by Atlassian.

Okay. We will look into why the CLA check is failing.

**Comment by jasontedor** [comment]
@detouched I think that we sorted out the CLA check issue, but I think that you need to leave a comment here to trigger the check to execute again. Do you mind giving that a try?

**Comment by detouched** [comment]
Sure =) Let's try it!

**Comment by detouched** [comment]
Awesome, it worked! Thank you @jasontedor!

**Comment by jasontedor** [comment]
Thanks @detouched! I'll step aside and let @Mpdreamz or @gmarz finish it from here. :smile:

**Comment by detouched** [comment]
I rebased my changes to current `master` to resolve the conflict.

**Comment by dakrone** [comment]
Ping @Mpdreamz or @gmarz want to merge this?

**Comment by gmarz** [comment]
@detouched would you mind rebasing and squashing to a single commit?  It'll make back-porting to 2.x a bit easier.

**Comment by detouched** [comment]
@gmarz Sure — rebased and squashed.

**Comment by gmarz** [comment]
Merged to master and cherry-picked to 2.x.  Thanks @detouched!

**Pull request merged**